apps/teams-test-app/src/App.tsx

```diff
@@ -16,29 +16,61 @@ const urlParams = new URLSearchParams(window.location.search);
 const getOriginsParam = urlParams.has('origins') && urlParams.get('origins') ? urlParams.get('origins') : '';
 const validMessageOrigins: string[] | undefined = getOriginsParam ? getOriginsParam.split(',') : undefined;
 
-// This is added for custom initialization when app can be initialized based upon a trigger/click.
-if (!urlParams.has('customInit') || !urlParams.get('customInit')) {
-  if (isTestBackCompat()) {
-    initialize(undefined, validMessageOrigins);
+if (!urlParams.has('testInit') || !urlParams.get('testInit')) {
+  // This is added for custom initialization when app can be initialized based upon a trigger/click.
+  if (!urlParams.has('customInit') || !urlParams.get('customInit')) {
+    if (isTestBackCompat()) {
+      initialize(undefined, validMessageOrigins);
+    } else {
+      app.initialize(validMessageOrigins);
+    }
+  }
+
+  // for AppInitialization tests we need a way to stop the Test App from sending these
+  // we do it by adding appInitializationTest=true to query string
+  if (
+    (urlParams.has('customInit') && urlParams.get('customInit')) ||
+    (urlParams.has(appInitializationTestQueryParameter) && urlParams.get(appInitializationTestQueryParameter))
+  ) {
+    console.info('Not calling appInitialization because part of App Initialization Test run');
   } else {
-    app.initialize(validMessageOrigins);
+    if (isTestBackCompat()) {
+      appInitialization.notifyAppLoaded();
+      appInitialization.notifySuccess();
+    } else {
+      app.notifyAppLoaded();
+      app.notifySuccess();
+    }
   }
 }
 
-// for AppInitialization tests we need a way to stop the Test App from sending these
-// we do it by adding appInitializationTest=true to query string
-if (
-  (urlParams.has('customInit') && urlParams.get('customInit')) ||
-  (urlParams.has(appInitializationTestQueryParameter) && urlParams.get(appInitializationTestQueryParameter))
-) {
-  console.info('Not calling appInitialization because part of App Initialization Test run');
-} else {
-  if (isTestBackCompat()) {
-    appInitialization.notifyAppLoaded();
-    appInitialization.notifySuccess();
-  } else {
-    app.notifyAppLoaded();
-    app.notifySuccess();
+window.addEventListener('message', handleMessageFromMockedHost);
+
+function handleMessageFromMockedHost(msg: MessageEvent): void {
+  if (!msg || !msg.data) {
+    console.log('Unrecognized message format received by app, message being ignored. Message: %o', msg);
+    return;
+  }
+  // Ensure the message is from a trusted origin
```

Comment:
How would you feel about writing a log statement for the cases where we are handling the message? It might be useful to be able to tell from the console logs that `app.initialize` was called because of this handler as opposed to some other reason.

Maybe the simplest way to do it would be something like 
```suggestion
console.log(`Received message from test host: ${msg.data}`);

  // Ensure the message is from a trusted origin
```